From 96f4d69496b483b8eb65dc4602d5cb96844f6647 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 30 Sep 2020 12:13:44 +0100
Subject: [PATCH] staging: bcm2835-codec: Correct buffer number
 change on start streaming

"cac8c90 staging: vc04_service: codec: Allow start_streaming to update
the buffernum" allowed the number of buffers configured to be decreased
as well as increased, but there is no requirement for all buffers to
have been queued when start_streaming is called.

Only allow increasing the buffernum.

Fixes: "cac8c90 staging: vc04_service: codec: Allow start_streaming to update the buffernum"
Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 .../staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
+++ b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
@@ -2290,7 +2290,7 @@ static int bcm2835_codec_start_streaming
 	if (count < port->minimum_buffer.num)
 		count = port->minimum_buffer.num;
 
-	if (port->current_buffer.num != count + 1) {
+	if (port->current_buffer.num < count + 1) {
 		v4l2_dbg(2, debug, &ctx->dev->v4l2_dev, "%s: ctx:%p, buffer count changed %u to %u\n",
 			 __func__, ctx, port->current_buffer.num, count + 1);
 
